lib/dogma/util/ast_node.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Dogma.Util.ASTNode do
+  @moduledoc"""
+  Utility functions for analyzing and categorizing AST nodes
+  """
+
+  @doc """
+  Returns true if the node x is a literal
+      
+      iex> Dogma.Util.ASTNode.literal?( "foo" )
+      true
+
+      iex> Dogma.Util.ASTNode.literal?({:foo, [], Elixir})
+      false
+  """
+  def literal?(x) when is_atom    x do true end
+  def literal?(x) when is_binary  x do true end
+  def literal?(x) when is_boolean x do true end
+  def literal?(x) when is_list    x do true end
+  def literal?(x) when is_number  x do true end
+  def literal?({_,_})               do true end
+  def literal?({:{}, _, _})         do true end
+  def literal?({:%{}, _, _})        do true end
+  def literal?(node) do
+    sigil?(node)
+  end
+
+  def sigil?({fun, _, _}) when is_atom fun do
```

Comment:
If we build functions for each of these by iterating over the letters of the alphabet we will check all sigils and have no runtime penalty. Additionally, this matches function calls like this:

```elixir
def sigil_transformer(x) do
  # ...
end
```

https://github.com/lpil/dogma/blob/0d0ece43081c28059e487bd0b681068c9d3ceb4a/lib/dogma/rules/quotes_in_string.ex#L23-L26